File: 0724-find-pivot-index/0724-find-pivot-index.java
class Solution {
    public int pivotIndex(int[] nums) {
        int n = nums.length,sum = 0,lsum=0;
        for(int i = 0; i<n; i++) sum += nums[i];
        for(int i = 0; i<n; i++){
            if(lsum == sum - lsum - nums[i]) return i;
            lsum += nums[i]; 
        }
        return -1;
    }
}

File: 0724-find-pivot-index/0724-find-pivot-index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int sum = 0,lsum = 0;
        for(int x : nums){
            sum += x;
        }
        for(int i=0; i<nums.size(); i++){
            int rsum = sum - nums[i] - lsum ;
            if(lsum == rsum) return i;
            lsum += nums[i];
        }
        return -1;
    }
};